src/components/timecard/add-timecard-period/AddTimeCardPeriod.jsx

```diff
@@ -1,14 +1,12 @@
-import { useTimecardContext } from '../../../context/TimecardContext';
 import { Link } from 'react-router-dom';
 import { addTimePeriodHeading } from '../../../utils/time-entry-utils/timeEntryUtils';
+import { PropTypes } from 'prop-types';
 
-const AddTimeCardPeriod = () => {
-  const { setSummaryErrors, setNewTimeEntry } = useTimecardContext();
-
+const AddTimeCardPeriod = ({ setSummaryErrors, setAddNewTimeEntry }) => {
   const handleAddTimePeriod = (event) => {
     event.preventDefault();
-    setSummaryErrors({});
-    setNewTimeEntry(true);
+    setSummaryErrors([]);
+    setAddNewTimeEntry(true);
```

Comment:
The control is now being made responsible for things outside of it's scope. I think this would be one event e.g. `onAddNewTimeEntry` that something else would respond to modify the state and clear errors etc.

Reply:
To be resolved in future work